File: ssm-sdk-client/src/main/java/ssm/client/repository/CommandArgs.java
package ssm.client.repository;

import java.util.List;
import java.util.Objects;

import com.google.common.base.MoreObjects;

import ssm.client.invoke.command.InvokeArgs;

public class CommandArgs {

    private String cmd;
    private String fcn;
    private List<String> args;

    public static CommandArgs from(String cmd, InvokeArgs invokeArgs, String channelId, String chaincodeId) {
        return new CommandArgs()
            .setCmd(cmd)
            .setFcn(invokeArgs.getFcn())
            .setArgs(invokeArgs.getArgs());
    }

    public String getCmd() {
        return cmd;
    }

    public CommandArgs setCmd(String cmd) {
        this.cmd = cmd;
        return this;
    }

    public String getFcn() {
        return fcn;
    }

    public CommandArgs setFcn(String fcn) {
        this.fcn = fcn;
        return this;
    }

    public List<String> getArgs() {
        return args;
    }

    public CommandArgs setArgs(List<String> args) {
        this.args = args;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof CommandArgs)) return false;
        CommandArgs commandArgs = (CommandArgs) o;
        return Objects.equals(cmd, commandArgs.cmd) &&
                Objects.equals(fcn, commandArgs.fcn) &&
                Objects.equals(args, commandArgs.args);
    }

    @Override
    public int hashCode() {
        return Objects.hash(cmd, fcn, args);
    }

    @Override
    public String toString() {
        MoreObjects.ToStringHelper stringHelper = MoreObjects.toStringHelper(this)
            .add("cmd", cmd)
            .add("fcn", fcn)
            .add("args", args);

        return stringHelper.toString();
    }
}

File: ssm-sdk/ssm-sdk-core/src/main/kotlin/ssm/chaincode/client/repository/CommandArgs.kt
package ssm.chaincode.client.repository

import java.util.Objects
import ssm.chaincode.client.model.InvokeArgs

class CommandArgs {
	var cmd: String? = null
		private set
	var fcn: String? = null
		private set
	var args: List<String>? = null
		private set

	fun setCmd(cmd: String?): CommandArgs {
		this.cmd = cmd
		return this
	}

	fun setFcn(fcn: String?): CommandArgs {
		this.fcn = fcn
		return this
	}

	fun setArgs(args: List<String>?): CommandArgs {
		this.args = args
		return this
	}

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is CommandArgs) return false
		val commandArgs = other
		return cmd == commandArgs.cmd &&
				fcn == commandArgs.fcn &&
				args == commandArgs.args
	}

	override fun hashCode(): Int {
		return Objects.hash(cmd, fcn, args)
	}

	override fun toString() = "cmd: $cmd, fcn: $fcn, args: $args"

	companion object {
		fun from(cmd: String?, invokeArgs: InvokeArgs, channelId: String?, chaincodeId: String?): CommandArgs {
			return CommandArgs()
				.setCmd(cmd)
				.setFcn(invokeArgs.fcn)
				.setArgs(invokeArgs.args)
		}
	}
}
